File: 프로그래머스/unrated/181862. 세 개의 구분자/세 개의 구분자.cpp
#include <string>
#include <vector>
#include <sstream>
using namespace std;

vector<string> solution(string myStr) {
    vector<string> answer;
    int count = 0;
    for (int i = 0; i < myStr.size(); i++)
    {
        if (myStr[i] == 'a' || myStr[i] == 'b' || myStr[i] == 'c')
        {
            myStr[i] = ' ';
            count++;
        }
    }
    if (count == myStr.size())
    {
        answer.push_back("EMPTY");
        return answer;
    }
    stringstream ss;
    ss.str(myStr);
    string word;
    while (ss >> word)
    {
        answer.push_back(word);
    }
    return answer;
}

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.java
import java.util.*;

class Solution {
    public String[] solution(String myStr) {
        String[] answer = Arrays.stream(myStr.split("[abc]"))
            .filter(s -> !s.isEmpty())
            .toArray(String[]::new);
        
        return answer.length == 0 ? new String[] {"EMPTY"} : answer;
    }
}